integration-cli/docker_cli_push_test.go

```diff
@@ -663,3 +664,55 @@ func (s *DockerRegistriesSuite) TestPushNeedsAuth(c *check.C) {
 		c.Fatalf("Wanted %s, got %s", expected, out)
 	}
 }
+
+func (s *DockerRegistrySuite) TestPushWithSkipSchema2(c *check.C) {
+	c.Assert(s.d.StartWithBusybox("--skip-schema2-push=true"), check.IsNil)
+
+	repo := fmt.Sprintf("%s/runcom/busybox", s.reg.url)
+	out, err := s.d.Cmd("tag", "busybox", repo)
+	c.Assert(err, check.IsNil, check.Commentf(out))
+
+	out, err = s.d.Cmd("push", repo)
+	c.Assert(err, check.IsNil, check.Commentf(out))
+
+	digest1 := reference.DigestRegexp.FindString(out)
+	c.Assert(len(digest1), checker.GreaterThan, 0, check.Commentf("no digest found for pushed manifest"))
+
+	out, err = s.d.Cmd("pull", repo+"@"+digest1)
+	c.Assert(err, check.IsNil, check.Commentf(out))
```

Comment:
That's right, you should see the error. But it's a possibility we've actually considered (to accept different digest than you receive). It's not critical since we decided not to do that. Leaving this up to you.